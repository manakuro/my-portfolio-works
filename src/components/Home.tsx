import * as React from 'react';
import { Animated } from 'react-animated-css';
import Anime, { AnimeProps } from 'react-anime';
import { CSSTransition } from 'react-transition-group';

import * as ReactMarkdown from 'react-markdown';

/* tslint:disable */

// components
import Header from './Header';
import Works from './Works';

// others
import '../App.css';
import sampleImg from '../static/images/sample.jpeg';
import addClassHtml from '../utils/addClassHtml';
import removeClassHtml from '../utils/removeClassHtml';
import { IHomeState as IAppProps } from '../reducers/home';

interface IAppState {
  offsetWidth: number;
  offsetHeight: number;
  isAnime: boolean;
  isShowOverlay: boolean;
  pageX: number;
  pageY: number;
  circleStyle: object;
  isShowWorksContent: boolean;
  isWorksContentAnimation: boolean;
  worksContent: string;
  img: string;
}

const WORKS_CONTENT = `
## EC Website

This is a copy website of NIKE.com, which is developed by Vue.js, 
Ruby on Rails and GraphQL.

Changes are automatically rendered as you type.

- Implements [GitHub Flavored Markdown](https://github.github.com/gfm/)
- Renders actual, "native" React DOM elements
- Allows you to escape or skip HTML (try toggling the checkboxes above)
- If you escape or skip the HTML, no \`dangerouslySetInnerHTML\` is used! Yay!

## HTML block below

<blockquote>
  This blockquote will change based on the HTML settings above.
</blockquote>

## How about some code?
\`\`\`js
var React = require('react');
var Markdown = require('react-markdown');

React.render(
  <Markdown source="# Your markdown here" />,
  document.getElementById('content')
);
\`\`\`

Pretty neat, eh?

## Tables?

| Feature | Support |
| ------ | ----------- |
| tables | ✔ |
| alignment | ✔ |
| wewt | ✔ |

## More info?

Read usage information and more on [GitHub](//github.com/rexxars/react-markdown)

---------------

A component by [VaffelNinja](http://vaffel.ninja) / Espen Hovlandsdal

## EC Website

This is a copy website of NIKE.com, which is developed by Vue.js, 
Ruby on Rails and GraphQL.

Changes are automatically rendered as you type.

- Implements [GitHub Flavored Markdown](https://github.github.com/gfm/)
- Renders actual, "native" React DOM elements
- Allows you to escape or skip HTML (try toggling the checkboxes above)
- If you escape or skip the HTML, no \`dangerouslySetInnerHTML\` is used! Yay!

## HTML block below

<blockquote>
  This blockquote will change based on the HTML settings above.
</blockquote>

## How about some code?
\`\`\`js
var React = require('react');
var Markdown = require('react-markdown');

React.render(
  <Markdown source="# Your markdown here" />,
  document.getElementById('content')
);
\`\`\`

Pretty neat, eh?

## Tables?

| Feature | Support |
| ------ | ----------- |
| tables | ✔ |
| alignment | ✔ |
| wewt | ✔ |

## More info?

Read usage information and more on [GitHub](//github.com/rexxars/react-markdown)

---------------

A component by [VaffelNinja](http://vaffel.ninja) / Espen Hovlandsdal
`;

class App extends React.Component<IAppProps, IAppState> {
  constructor(props: IAppProps) {
    super(props);
    this.showOverlay = this.showOverlay.bind(this);
    this.showWorksContent = this.showWorksContent.bind(this);
    this.hideOverlay = this.hideOverlay.bind(this);
    this.onExitedOverlay = this.onExitedOverlay.bind(this);

    this.state = {
      offsetWidth: 0,
      offsetHeight: 0,
      isAnime: false,
      isShowOverlay: false,
      pageX: 0,
      pageY: 0,
      circleStyle: {},
      isShowWorksContent: false,
      isWorksContentAnimation: false,
      worksContent: '',
      img: sampleImg,
    };
  }

  async showOverlay(e: React.SyntheticEvent<EventTarget>) {
    const nativeEvent: any = e.nativeEvent;
    const { pageX, pageY } = nativeEvent;
    await this.setState({
      isShowOverlay: true,
      pageX,
      pageY,
      circleStyle: {
        top: `${pageY - 50}px`,
        left: `${pageX - 50}px`,
      },
    });
  }

  hideOverlay(): void {
    this.setState({
      isShowWorksContent: false,
    });
    removeClassHtml();

    this.setState({
      isShowOverlay: false,
      isWorksContentAnimation: false,
    });
  }

  onExitedOverlay(): void {
    this.setState({
      circleStyle: {
        top: '0px',
        left: '-1000px',
      },
    });
  }

  componentDidMount(): void {}

  showWorksContent(): void {
    this.setState({
      isShowWorksContent: true,
      worksContent: WORKS_CONTENT,
    });
    addClassHtml('hidden');
  }

  onEnteredShowWorksContent(): void {
    this.setState({
      isWorksContentAnimation: true,
    });
  }

  public render(): JSX.Element {
    const animeProps: AnimeProps = {
      children: [],
      easing: 'easeInOutCirc',
      duration: 1000,
      delay: (_, i: number, t: number) => Math.abs(t / 2 - i) * 80,
      translateX: '-100%',
      scale: 1,
      value: '',
      translateY: 0,
      rotate: 0,
      opacity: 1,
      color: '',
      backgroundColor: '',
      points: '',
      strokeDashoffset: 0,
    };

    return (
      <div className="App">
        <CSSTransition
          in={this.state.isShowWorksContent}
          classNames="slide-works"
          timeout={1000}
          onEntered={() => this.onEnteredShowWorksContent()}
        >
          <div className="works-content-overlay">
            <Animated
              animationIn="fadeInDown"
              animationOut="fadeOut"
              isVisible={this.state.isWorksContentAnimation}
            >
              <h2 className="works-content-heading">EC Website</h2>
            </Animated>
            <Animated
              animationIn="fadeInDown"
              animationOut="fadeOut"
              animationInDelay={300}
              isVisible={this.state.isWorksContentAnimation}
            >
              <ReactMarkdown
                className="markdown-body"
                source={this.state.worksContent}
              />
            </Animated>
          </div>
        </CSSTransition>
        <Animated
          className="works-content-left-wrapper"
          animationIn="fadeIn"
          animationOut="fadeOut"
          animateOnMount={false}
          isVisible={this.state.isWorksContentAnimation}
        >
          <div className="works-content-left">
            <div className="works-content-left-inner">
              <Animated
                animationIn="fadeInDown"
                animationOut="fadeOut"
                animationInDelay={300}
                isVisible={this.state.isWorksContentAnimation}
              >
                <div className="uncover">
                  <a href="https://google.com" target="blank">
                    <img src={this.state.img} />
                  </a>

                  <div className="uncover-slices">
                    {false && (
                      <Anime {...animeProps}>
                        <div className="uncover-slice" />
                        <div className="uncover-slice" />
                        <div className="uncover-slice" />
                        <div className="uncover-slice" />
                        <div className="uncover-slice" />
                        <div className="uncover-slice" />
                      </Anime>
                    )}
                  </div>
                </div>
              </Animated>
            </div>
          </div>
        </Animated>

        <CSSTransition
          in={this.state.isShowOverlay}
          classNames="scale"
          timeout={1000}
          onEnter={() => this.showWorksContent()}
          onExit={() => this.onExitedOverlay()}
        >
          <div className="circle" style={this.state.circleStyle} />
        </CSSTransition>

        <CSSTransition
          in={this.state.isShowWorksContent}
          classNames="slide-button"
          timeout={1000}
        >
          <div className="works-back-button">
            <i className="fa fa-angle-left icon" onClick={this.hideOverlay} />
          </div>
        </CSSTransition>

        <Header />
        <main className="main container">
          <Works showOverlay={this.showOverlay} {...this.props} />
        </main>
      </div>
    );
  }
}

export default App;
