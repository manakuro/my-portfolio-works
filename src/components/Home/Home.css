.home {
  min-height: 100vh; }

.works {
  margin-top: 130px;
  margin-bottom: 50px; }
  .works .works-list {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-gap: 15px; }
