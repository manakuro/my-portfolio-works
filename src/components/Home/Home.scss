$headerHeight: 100px;
$bgOverlay: rgb(244, 247, 246);

.works {
  $marginTop: $headerHeight + 30px;

  margin-top: $marginTop;
  margin-bottom: 50px;

  .works-list {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-gap: 15px;
  }
}

.overlay {
  width: 100vw;
  height: 100vh;
  position: fixed;
  z-index: 2001;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.circle {
  border-radius: 50%;
  background: rgb(244, 247, 246);
  width: 100px;
  height: 100px;
  position: absolute;
  left: -1000px;
  top: 0;
  z-index: 2001;
  opacity: 0;
}

/* ------------------------------
  Works Content
 --------------------------------*/
.works-content-overlay {
  width: 50%;
  height: 100%;
  position: fixed;
  left: 50%;
  top: 0;
  z-index: 3001;
  transform: translateX(100%);
  background: #fff;
  padding: 0 30px 30px 30px;
  overflow-y: scroll;

  .works-content-heading {
    margin-top: 80px;
    margin-bottom: 80px;
    font-size: 4rem;
    line-height: 1.5;
  }
}

// enter transition
.slide-works-enter-active {
  opacity: 1;
  transform: translateX(0);
  transition: transform .5s cubic-bezier(0.215, 0.610, 0.355, 1.000) .4s;
}
.slide-works-enter-done, .slide-works-exit {
  opacity: 1;
  transform: translateX(0);
}

// exit transition
.slide-works-exit-active {
  opacity: 0;
  transform: translateX(100%);
  transition: transform .7s cubic-bezier(0.215, 0.610, 0.355, 1.000), opacity .7s;
}

// back button
.works-back-button {
  position: fixed;
  top: 25px;
  left: 25px;
  z-index: 3001;
  transform: translateX(-100px);

  .icon {
    font-size: 2rem;
    cursor: pointer;
    width: 60px;
    height: 60px;
    background: #fff;
    border-radius: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
  }
}

.slide-button-enter-active {
  opacity: 1;
  transform: translateX(0);
  transition: transform .5s cubic-bezier(0.215, 0.610, 0.355, 1.000) .4s;
}

.slide-button-enter-done, .slide-button-exit {
  opacity: 1;
  transform: translateX(0);
}

.slide-button-exit-active {
  opacity: 1;
  transform: translateX(-100px);
  transition: transform .7s cubic-bezier(0.215, 0.610, 0.355, 1.000), opacity .7s;
}

/* ------------------------------
  Animation
 --------------------------------*/
.scale-enter-active {
  opacity: 1;
  transform: scale(40);
  transition: transform 2s cubic-bezier(0.215, 0.610, 0.355, 1.000) 0s; // easeOutCubic
}

.scale-enter-done, .scale-exit {
  opacity: 1;
  transform: scale(40);
}

.scale-exit-active {
  opacity: 0;
  transition: opacity .5s;
}


/* ------------------------------
  Works content left
 --------------------------------*/
.works-content-left-wrapper {
  position: relative;
  z-index: 3001;
  opacity: 0;
}

.works-content-left {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50%;
  height: 100%;
  position: fixed;
  left: 0;
  top: 0;
  z-index: 3001;
  background: transparent;
  padding: 0 50px;
}

.works-content-left-inner {
  width: 90%;
}

/* ------------------------------
  Uncover
 --------------------------------*/
.uncover {
  overflow: hidden;
  background-image: none !important;
  position: relative;

  .uncover-slices {
    flex-direction: column;
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    display: flex;

    .uncover-slice {
      flex: 1;
      color: $bgOverlay;
      background: currentColor;
    }
  }
}


/* ------------------------------
  Overwrite github-markdown-css
 --------------------------------*/
.markdown-body {
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  color: #333;

  ul, li {
    list-style: initial;
  }
}



