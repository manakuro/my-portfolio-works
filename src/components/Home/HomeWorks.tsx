import * as React from 'react';

import { IHomeState, IWork } from '@/reducers/home';

interface IWorksProps extends IHomeState {
  showOverlay: (e: React.SyntheticEvent<EventTarget>) => void;
}

export default class Works extends React.Component<IWorksProps, {}> {
  constructor(props: IWorksProps) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  public render(): JSX.Element {
    return (
      <div className="works">
        <ul className="works-list">
          {this.props.works.map((work: IWork, i: number) => (
            <li className="works-list-item" key={i} onClick={this.onClick}>
              <div className="card">
                <img src={work.img} />

                <div className="card-desc">
                  <h3 className="card-desc-heading">{work.title}</h3>
                  <p className="card-desc-sub-heading">{work.description}</p>

                  <ul className="tech-list">
                    <li className="tech-list-item">
                      <span className="icon-vuejs">
                        <span className="path1" />
                        <span className="path2" />
                      </span>
                    </li>
                    <li className="tech-list-item">
                      <span className="icon-vuejs">
                        <span className="path1" />
                        <span className="path2" />
                      </span>
                    </li>
                  </ul>
                </div>
              </div>
            </li>
          ))}
        </ul>
      </div>
    );
  }

  private onClick(e: any) {
    this.props.showOverlay(e);
  }
}
