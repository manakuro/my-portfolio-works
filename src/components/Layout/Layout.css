.layout {
  position: relative; }

.loader {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background: #fff; }

.loader-enter-active {
  opacity: 0;
  visibility: visible;
  transition: .5s; }

.loader-enter-done, .loader-exit {
  opacity: 0;
  visibility: hidden; }

.loader-exit-active {
  opacity: 1;
  visibility: visible; }
