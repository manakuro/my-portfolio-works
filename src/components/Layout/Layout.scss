.layout {
  position: relative;
}

.loader {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background: #fff;
}


.loader-enter-active {
  opacity: 0;
  transition: .5s;
}

.loader-enter-done, .loader-exit {
  opacity: 0;
}

.loader-exit-active {
  opacity: 1;
}
