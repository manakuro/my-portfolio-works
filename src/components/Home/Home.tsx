import * as React from 'react'
import { connect, Dispatch } from 'react-redux'

// others
import './Home.css'
import { HomeState, Work, SearchQuery } from '@/modules/home/reducer'
import actions from '@/modules/home/actions'
import { RootState } from '@/modules/reducers'
import WorkListItem from '@/components/WorkListItem/WorkListItem'
import { History, Location } from 'history'
import * as queryString from 'query-string'
import { WorkDetail } from '@/components/WorkDetail/WorkDetail'
import { Route } from 'react-router'

export interface HomeProps extends HomeStateFromProps, HomeDispatchFromProps {}

export class Home extends React.Component<HomeProps, {}> {
  constructor(props: HomeProps) {
    super(props)
  }

  public async componentWillMount(): Promise<void> {
    await this.mergeQuery()
    this.props.fetchWorks()
  }

  public componentDidUpdate(prevProps: HomeProps) {
    if (prevProps.searchQuery !== this.props.searchQuery)
      this.props.fetchWorks()
  }

  public render(): JSX.Element {
    return (
      <div className="home">
        <Route
          path="/works/:id"
          exact={true}
          render={routeProps => {
            // tslint:disable-next-line jsx-no-lambda
            return <WorkDetail {...this.props} {...routeProps} />
          }}
        />
        <div className="works">
          <div className="works-list">
            {this.props.works.map((work, index) => (
              <WorkListItem
                key={index}
                work={work}
                updateTargetWork={this.props.updateTargetWork}
                languages={this.props.languages}
                updateCircle={this.props.updateCircle}
                history={this.props.history}
              />
            ))}
          </div>
        </div>
      </div>
    )
  }

  private async mergeQuery(): Promise<void> {
    if (this.props.location) {
      const query = queryString.parse(this.props.location.search)
      if (query.languages) {
        if (typeof query.languages === 'string')
          query.languages = query.languages.split()
        query.languages = query.languages.map((l: string) => parseInt(l, 10))
      }

      await this.props.updateSearchQuery(query)
    }
  }
}

export interface HomeDispatchFromProps {
  toggleOverlay: (isShowOverlay: boolean) => any
  toggleWorksContent: (isShowWorksContent: boolean) => any
  toggleWorksContentAnimation: (isShowWorksContentAnimation: boolean) => any
  toggleWorksContentExpand: (isExpandWorksContent: boolean) => any
  updateCircle: (circleStyle: React.CSSProperties) => any
  updateWorkContentImg: (workContentImg: string) => any
  updateTargetWork: (payload: Work) => any
  updateSearchQuery: (searchQuery: any) => any
  fetchWorks: (searchQuery?: SearchQuery) => any
  fetchWork: (id: number) => any
}

export interface HomeStateFromProps extends HomeState {
  location?: Location
  history?: History
}

export function mapStateToProps(state: RootState) {
  return {
    ...state.home,
  }
}

export function mapDispatchToProps(dispatch: Dispatch<() => any>) {
  return {
    toggleOverlay: (isShowOverlay: boolean) =>
      dispatch(actions.toggleOverlay(isShowOverlay)),
    toggleWorksContent: (isShowWorksContent: boolean) =>
      dispatch(actions.toggleWorksContent(isShowWorksContent)),
    toggleWorksContentAnimation: (isShowWorksContentAnimation: boolean) =>
      dispatch(
        actions.toggleWorksContentAnimation(isShowWorksContentAnimation),
      ),
    toggleWorksContentExpand: (isExpandWorksContent: boolean) =>
      dispatch(actions.toggleWorksContentExpand(isExpandWorksContent)),

    updateCircle: (circleStyle: React.CSSProperties) =>
      dispatch(actions.updateCircle(circleStyle)),
    updateWorkContentImg: (workContentImg: string) =>
      dispatch(actions.updateWorkContentImg(workContentImg)),
    updateTargetWork: (targetWork: Work) =>
      dispatch(actions.updateTargetWork(targetWork)),
    updateSearchQuery: (searchQuery: any) =>
      dispatch(actions.updateSearchQuery(searchQuery)),

    fetchWorks: () => dispatch(actions.fetchWorks.request()),
    fetchWork: (id: number) => dispatch(actions.fetchWork.request(id)),
  }
}

export default connect<HomeStateFromProps, HomeDispatchFromProps, void>(
  mapStateToProps,
  mapDispatchToProps,
)(Home)
