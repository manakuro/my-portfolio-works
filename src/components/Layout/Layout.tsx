import * as React from 'react'
import { connect, Dispatch } from 'react-redux'
import { CSSTransition } from 'react-transition-group'

// components
import Header from '@/components/Header/Header'

// others
import './Layout.css'
import { HomeState } from '@/modules/home/reducer'
import actions from '@/modules/home/actions'
import { RouteComponentProps } from 'react-router'
import { RootState } from '@/modules/reducers'

interface LayoutProps
  extends LayoutStateFromProps,
    LayoutDispatchFromProps,
    LayoutOwnProps {}

export class Layout extends React.Component<LayoutProps, {}> {
  constructor(props: LayoutProps) {
    super(props)
  }

  public async componentWillMount(): Promise<void> {
    this.props.fetchLanguages()
  }

  public render(): JSX.Element {
    return (
      <div className="layout">
        <Header {...this.props} />

        <CSSTransition
          in={!this.props.isLoading}
          timeout={200}
          classNames="loader"
        >
          <div className="loader" />
        </CSSTransition>

        <main className="main container">{this.props.children}</main>
      </div>
    )
  }
}

export interface LayoutDispatchFromProps {
  fetchLanguages: () => any
  updateSearchQuery: (searchQuery: any) => any
}

export interface LayoutStateFromProps extends HomeState {}

export interface LayoutOwnProps extends RouteComponentProps<any> {}

export function mapStateToProps(state: RootState, ownProps: LayoutOwnProps) {
  return {
    ...state.home,
    ...ownProps,
  }
}

export function mapDispatchToProps(dispatch: Dispatch<() => any>) {
  return {
    fetchLanguages: () => dispatch(actions.fetchLanguages.request()),
    updateSearchQuery: (searchQuery: any) =>
      dispatch(actions.updateSearchQuery(searchQuery)),
  }
}

export default connect<
  LayoutStateFromProps,
  LayoutDispatchFromProps,
  LayoutOwnProps
>(mapStateToProps, mapDispatchToProps)(Layout) as any // @todo resolve types
