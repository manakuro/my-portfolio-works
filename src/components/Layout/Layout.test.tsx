import * as React from 'react'
import { shallow } from 'enzyme'
import * as Layout from '@/components/Layout/Layout'
import * as Home from '@/components/Home/Home'

const propsData: Layout.LayoutProps = {
  works: [
    {
      id: 1,
      title: 'EC Website',
      description: `
      Iâ€™ve been a CMT for ten years now, and I have, literally and
      figuratively, held the pulse of a steaming cross-section of
      San Franciscans
      `,
      img: 'http://sample.jpg',
      component: 'WorkOne',
      languages: [1, 2],
    },
  ],
  workContentImg: 'http://sample.jpg',
  isShowOverlay: false,
  isShowWorksContent: false,
  isShowWorksContentAnimation: false,
  isExpandWorksContent: false,
  circleStyle: {
    top: 'calc(50% - 50px)',
    left: 'calc(50% - 50px)',
  },
  targetWork: null,
  languages: [],
  searchQuery: {
    languages: [1, 2],
  },
  isLoading: true,
  history: {
    push: jest.fn(),
  } as any,
  match: {
    params: {
      id: 1,
    },
  } as any,
  location: {
    search: {
      query: {
        languages: '1',
      },
    },
  } as any,

  updateSearchQuery: jest.fn(),
  fetchLanguages: jest.fn(),
}

describe('Layout', () => {
  let props: Layout.LayoutProps
  beforeEach(() => {
    props = { ...propsData }
  })

  it('should render component', () => {
    const wrapper = shallow(<Layout.Layout {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('componentWillMount', () => {
    let component: any
    beforeEach(() => {
      component = new Layout.Layout(props)
    })

    it('should call fetchLanguages', async () => {
      props.fetchLanguages = jest.fn()

      await component.componentWillMount()

      expect(props.fetchLanguages).toHaveBeenCalled()
    })
  })
})
