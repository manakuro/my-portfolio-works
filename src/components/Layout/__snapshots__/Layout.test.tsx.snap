// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render component 1`] = `
<div
  className="layout"
>
  <Header
    circleStyle={
      Object {
        "left": "calc(50% - 50px)",
        "top": "calc(50% - 50px)",
      }
    }
    fetchLanguages={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    isExpandWorksContent={false}
    isLoading={true}
    isShowOverlay={false}
    isShowWorksContent={false}
    isShowWorksContentAnimation={false}
    languages={Array []}
    location={
      Object {
        "search": Object {
          "query": Object {
            "languages": "1",
          },
        },
      }
    }
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    searchQuery={
      Object {
        "languages": Array [
          1,
          2,
        ],
      }
    }
    targetWork={null}
    updateSearchQuery={[MockFunction]}
    workContentImg="http://sample.jpg"
    works={
      Array [
        Object {
          "component": "WorkOne",
          "description": "
      Iâ€™ve been a CMT for ten years now, and I have, literally and
      figuratively, held the pulse of a steaming cross-section of
      San Franciscans
      ",
          "id": 1,
          "img": "http://sample.jpg",
          "languages": Array [
            1,
            2,
          ],
          "title": "EC Website",
        },
      ]
    }
  />
  <CSSTransition
    classNames="loader"
    in={false}
    timeout={200}
  >
    <div
      className="loader"
    />
  </CSSTransition>
  <main
    className="main container"
  />
</div>
`;
